src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeMemory.Windows.cs

```diff
@@ -0,0 +1,199 @@
+        {
+            nuint byteCount = GetByteCount(elementCount, elementSize);
+            return Alloc(byteCount);
+        }
+
+        /// <summary>Allocates and zeroes a block of memory of the specified size, in bytes.</summary>
+        /// <param name="byteCount">The size, in bytes, of the block to allocate.</param>
+        /// <returns>A pointer to the allocated and zeroed block of memory.</returns>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="byteCount" /> of memory failed.</exception>
+        /// <remarks>
+        ///     <para>This method allows <paramref name="byteCount" /> to be <c>0</c> and will return a valid pointer that should not be dereferenced and that should be passed to free to avoid memory leaks.</para>
+        ///     <para>This method is a thin wrapper over the C <c>calloc</c> API.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void* AllocZeroed(nuint byteCount)
+        {
+            return AllocZeroed(byteCount, elementSize: 1);
+        }
+
+        /// <summary>Allocates and zeroes a block of memory of the specified size, in elements.</summary>
+        /// <param name="elementCount">The count, in elements, of the block to allocate.</param>
+        /// <param name="elementSize">The size, in bytes, of each element in the allocation.</param>
+        /// <returns>A pointer to the allocated and zeroed block of memory.</returns>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="elementCount" /> * <paramref name="elementSize" /> bytes of memory failed.</exception>
+        /// <remarks>
+        ///     <para>This method allows <paramref name="elementCount" /> and/or <paramref name="elementSize" /> to be <c>0</c> and will return a valid pointer that should not be dereferenced and that should be passed to free to avoid memory leaks.</para>
+        ///     <para>This method is a thin wrapper over the C <c>calloc</c> API.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void* AllocZeroed(nuint elementCount, nuint elementSize)
+        {
+            // The Windows implementation handles num == 0 && size == 0 as we expect
+            void* result = Interop.Ucrtbase.calloc(elementCount, elementSize);
+
+            if (result == null)
+            {
+                ThrowHelper.ThrowOutOfMemoryException();
+            }
+
+            return result;
+        }
+
+        /// <summary>Frees a block of memory.</summary>
+        /// <param name="ptr">A pointer to the block of memory that should be freed.</param>
+        /// <remarks>
+        ///    <para>This method does nothing if <paramref name="ptr" /> is <c>null</c>.</para>
+        ///    <para>This method is a thin wrapper over the C <c>free</c> API.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void Free(void* ptr)
+        {
+            Interop.Ucrtbase.free(ptr);
```

Comment:
Add null check to match the Unix version?

Reply:
Fixed.